src/PageHeader/PageHeader.tsx

```diff
@@ -219,16 +279,18 @@ const TitleArea: React.FC<React.PropsWithChildren<TitleAreaProps>> = ({
   return (
     <TitleAreaContext.Provider value={{titleVariant: currentVariant, titleAreaHeight: height}}>
```

Comment:
One thing I realised when a few jest tests failed is that since`PageHeader.LeadingAction` and `PageHeader.TrailingAction` are not listed in the PageHeader.TitleArea (please see the storybook diff) anymore, they don't have access to `PageHeader.TitleArea`s context :/ So when the variant is large, it doesn't have any impact on the leading and trailing action. One solution I can think of is lifting the state up to the PageHeader component, we can rename the prop as well `titleVariant` but let me know if you have any suggestions 🙌🏻

Reply:
Does it make sense to suggest to have a contextprovider only at pageheader level? Seems a bit granular to have it for a section.

Reply:
I think it is an option. When we lift the context provider to the parent component level, the props needs to be passed on that level too though, right? 

Instead of

```
<PageHeader>
    <PageHeader.TitleArea variant="large">
        <PageHeader.Title>This is the main title </PageHeader.Title>
    </PageHeader.TitleArea>
</PageHeader>
```

we will need to have 

```
<PageHeader variant="large">
    <PageHeader.TitleArea >
        <PageHeader.Title>This is the main title </PageHeader.Title>
    </PageHeader.TitleArea>
</PageHeader>
```

Because the variant is only relevant to the title area, this is why we kept the context in a granular level and only in the titlearea. 

If moving the context provider higher to `PageHeader` level and still making the prop on the `PageHeader.TitleArea` component is possible, let me know!

If we need to move the props to the parent level, we could rename the prop name as well maybe like `titleVariant`  or something?

Reply:
I think we can do this just with CSS and remove the React Context altogether. We can pass `data-size-variant={variant}` on the parent, then pass `data-component="ComponentName"` attribute to `PageHeader.LeadingAction`, `PageHeader.TrailingAction`, and `PageHeader.Title`  do something like this:

```
const StyledRoot = styled(Box)`
&[data-size-variant="small"] {

  [data-component="LeadingAction"] {
    /* small leading action variant styles go here */
  }

  [data-component="TrailingAction"] {
    /* small trailing action variant styles go here */
  }

  [data-component="Title"] {
    /* small title variant styles go here */
  }

}

/* repeat the above for every size variant with special styles */

`
```

We can pair on this if you want

Reply:
If `PageHeader.TitleArea` will always be a direct child of `PageHeader`, I think we can do this just with CSS and remove the React Context altogether.

- First, we need to grab the value of the `variant` prop from the `PageHeader.TitleArea`
- Then, we can pass `data-size-variant={variant}` on to the `Root` element.
- Then pass `data-component="ComponentName"` attribute to `PageHeader.LeadingAction`, `PageHeader.TrailingAction`, and `PageHeader.Title`  do something like this:

```
const StyledRoot = styled(Box)`
&[data-size-variant="small"] {

  [data-component="LeadingAction"] {
    /* small leading action variant styles go here */
  }

  [data-component="TrailingAction"] {
    /* small trailing action variant styles go here */
  }

  [data-component="Title"] {
    /* small title variant styles go here */
  }

}

/* repeat the above for every size variant with special styles */
`
```

We can pair on this if you want. I tried to think of a way to handle all of this in CSS without needing to know the variant, but I couldn't figure it out.

Reply:
I love this! 💖  I gave this a go and it made me realise that we actually don't need `variant` value in leading/trailing visuals/actions (I _think_). Because we use the `variant` value to set a height for the leading/trailing visuals/actions (2rem for `medium` and `subtitle` variant and 3rem for `large` variant) and it is only because of needing to align these actions and visuals with the title when it has variant font-size due to the `variant` prop. Setting `align-items: 'center'` in the parent page header grid resolved the alignment issue. I still use the data attributes as you suggested here to set the styles for the `Title` component and it is a great idea rather than dynamically rendering styles when the prop is changed. 

You can see the changes in [this commit](https://github.com/primer/react/pull/3816/commits/a6e2335ae4df9a65ccc3c32c425cd05351b8b4bf) and the [playground story](https://primer-402ed3ebc8-13348165.drafts.github.io/storybook/?path=/story/drafts-components-pageheader--playground) is a good one to test. Let me know what you think and if I misunderstand anything!

Reply:
> Setting align-items: 'center' in the parent page header grid resolved the alignment issue.

@broccolinisoup - setting `align-items: center` won't work for multiple lines of text. The leading/trailing visuals will be aligned to the vertical center of all lines of text, not just the first line of text.

Reply:
Oo I totally missed that the leading/trailing visuals need to be aligned with the first line of text 🤦‍♀️ Sorry, yeah that won't work. I'll also update the VRT tests to include these cases.